tasks/model.go

```diff
@@ -274,7 +275,36 @@ func (t *_Task) Finalize(ctx context.Context, s ipld.Storer) (FinishedTask, erro
 }
 
 func parseFinalLogs(t Task) (int, string, string, _Int__Maybe, _Int__Maybe, _Int__Maybe) {
-	return 0, "", "", _Int__Maybe{}, _Int__Maybe{}, _Int__Maybe{}
+	timeFirstByte := _Int__Maybe{m: schema.Maybe_Absent}
+	timeLastByte := _Int__Maybe{m: schema.Maybe_Absent}
+
+	start := t.StartedAt.Must().Time()
+	for _, stage := range t.PastStageDetails.Must().x {
+		for _, log := range stage.Logs.x {
+			entry := log.Log.String()
+			entryTime := log.UpdatedAt.Time()
+
+			// If funds are reserved, we're about to transfer the data.
+			// TODO: can we do better, e.g. the first transfer
+			// progress update?
+			// TODO: test for retrieval tasks, too.
+			if strings.Contains(entry, "funds reserved") && !timeFirstByte.Exists() {
```

Comment:
nit: Test `!timeFirstByte.Exists()` first, before doing more expensive `string.Contains`

Reply:
Sure, though I doubt this will matter in practice :)